test/functional/android/ime_tests.py

```diff
@@ -45,7 +44,8 @@ def test_is_ime_active(self):
         self.assertTrue(self.driver.is_ime_active())
 
     def test_active_ime_engine(self):
-        self.assertIsInstance(self.driver.active_ime_engine, unicode)
+        expected = str if is_py3() else unicode
+        self.assertIsInstance(self.driver.active_ime_engine, expected)
```

Comment:
```
try:
  basestring
except NameError:
  basestring = str

isinstance(s, basestring)
```

Reply:
Thanks, it's cool.

Reply:
^ unfortunately, below exception occurs on py2, so current impl is revised.

```
UnboundLocalError: local variable 'basestring' referenced before assignment
```